build/params_mainnet.go

```diff
@@ -126,3 +127,7 @@ const BootstrapPeerThreshold = 4
 
 // we skip checks on message validity in this block to sidestep the zero-bls signature
 var WhitelistedBlock = MustParseCid("bafy2bzaceapyg2uyzk7vueh3xccxkuwbz3nxewjyguoxvhx77malc2lzn2ybi")
+
+// CBDeliveryDelay is the delay before deliver in the synchronous consistent broadcast.
+// This determines the wait time for the detection of potential equivocations.
+var CBDeliveryDelay = 6 * time.Second
```

Comment:
> Does this impact block propagation? I guess no since pubsub checks don't touch this logic.

It shouldn't, right? The only thing we are delaying is the delivery of the block to the syncer (i.e. the call to [`InformNewBlock`](https://github.com/filecoin-project/lotus/pull/9858/files#diff-2b53d4ebd3dc2ee0318d15b3d746a581c19c2d376cf1bea25e309dc61214cfafR124)), but this is done after the Gossipsub validation so the propagation of the block should follow its course seamlessly. 

> Can we set this to a lower value while still getting the expected security benefits?

This is a great point, and this could definitely be a problem. These 6 seconds is the theoretical time that it takes for a block to be propagated to all of the network by Gossipsub. The assumption is that after this time there is high probability that any equivocation has been propagated enough to be caught by a majority of miners. Lowering this value will impact the effectiveness of the scheme, but I don't know to what extent or what is the minimum threshold. I know @jsoares and @guy.goren were working on getting some real data from mainnet with ProbeLab to see empirically what is the optimal delay. I'll let them chime in in case they have more info.

Reply:
> Does this impact block propagation? I guess no since pubsub checks don't touch this logic.

As @adlrocha said, propagation shouldn't be affected. Anything else would negate the benefits of the patch: we want the guard time to be larger than the propagation time, which is easier the faster propagation is.

> Can we set this to a lower value while still getting the expected security benefits?

The value can be set lower. In fact, there is no "cliff" behaviour and the security gains can be approximated (if not formally, at least intuitively) as a logistic function centred on the average propagation delay: while you don't necessarily _need_ to cover the 99th-percentile, too low thresholds would catch few attacks and you'd always want to at least cover the average case.

Alas, our only real-world data was collected from the bootstrapper nodes (via Sentinel). [Our analysis of that data](https://gist.github.com/jsoares/9ce4c0ba6ebcfd2afa8f8993890e2d98) shows that the vast majority of blocks reach the bootstrappers in < 1s. However, there are a few weaknesses here, primarily as (i) we're trusting source timestamps; (ii) bootstrapper nodes are not necessarily representative of the average node in terms of connectivity (both link & centrality, even mesh configuration). 

In the meantime, ProbeLab is working on a network-wide GossipSub study that will yield a higher-confidence distribution for block propagation. Alas, that work is taking longer than we expected and there is still no time frame for the results. The plan was, therefore, to start with the 6s used as the assumption in the GossipSub security analysis, keep it user-configurable, and push better defaults once we have the data. I expect we'll land in the 1 to 2 second range, but don't hold me to that.

Reply:
Note that here I'm more concerned about a block arriving late making SPs be late with their own block.

Propagation [delay is 10s](https://github.com/filecoin-project/lotus/blob/master/build/params_mainnet.go#L98), state compute today takes up to 14s due to cron issues. With this patch worst case if some set of SPs choose to send blocks late, we'd have essentially no time spare for WindowPoSt, essentially cutting into propagation delay for the next epoch - which afaict would be either comparable or worse than epoch boundary

Reply:
Right, that's a valid concern given we're right at the threshold. How high would you be comfortable going? Given the context above, I think we could conceivably adjust down a second or two to leave some buffer time -- and still reassess when the study results are available.